projects/exui-common-lib/src/lib/components/write-address/write-address.component.ts

```diff
@@ -0,0 +1,158 @@
+  }
+
+  public ngOnInit(): void {
+    if (!this.formGroup.get('address')) {
+      this.formGroup.addControl('address', new FormControl({}));
+    }
+    this.ukInternationalFormGroup = new FormGroup({
+      ukAddress: new FormControl()
+    });
+    this.postcode = new FormControl('');
+    this.addressFormGroup.addControl('postcode', this.postcode);
+    this.addressList = new FormControl('');
+    this.addressFormGroup.addControl('address', this.addressList);
+    this.ukAddress = new FormControl('');
+    this.addressFormGroup.addControl('ukAddress', this.ukAddress);
+  }
+
+  public findAddress() {
+    if (!this.postcode.value) {
+      this.missingPostcode = true;
+    } else {
+      this.missingPostcode = false;
+      this.addressField = null;
+      const postcode = this.postcode.value;
+      this.addressOptions = [];
+      this.addressesService.getAddressesForPostcode(postcode.replace(' ', '').toUpperCase()).subscribe(
+        result => {
+          result.forEach(
+            address => {
+              this.addressOptions.push(new AddressOption(address, null));
+            }
+          );
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        }, (error) => {
+          console.log(`An error occurred retrieving addresses for postcode ${postcode}. ${error}`);
+          this.addressOptions.unshift(
+            new AddressOption(undefined, this.defaultLabel(this.addressOptions.length))
+          );
+        });
+      this.addressList.setValue(undefined);
+      // this.refocusElement();
+    }
+  }
+
+  public blankAddress() {
+    this.setFormValue();
+    if (this.internationalMode) {
+      this.internationalModeStart.emit();
+    }
+  }
+
+  public shouldShowDetailFields() {
+    if (this.isExpanded) {
+      return true;
+    }
+    if (!this.formGroup.get('address')) {
+      return false;
+    }
+    const address = this.formGroup.get('address').value;
+    let hasAddress = false;
+    if (address) {
+      Object.keys(address).forEach((key) => {
+        if (address[key] !== null) {
+          hasAddress = true;
+        }
+      });
+    }
+    return hasAddress;
+  }
+
+  public addressSelected() {
+    this.addressField = this.addressList.value;
+    this.addressChosen = true;
+    this.setFormValue();
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
+    const change = changes['addressField'];
+    if (change) {
+      this.setFormValue();
+    }
+  }
+
+  public setInternationalAddress(event: any): void {
+    const target = event.target;
+    if (target.checked) {
+      this.ukRadioChecked = true;
+      this.isInternational = target.id === 'no';
+    }
+  }
+
+  private defaultLabel(numberOfAddresses: number) {
+    return numberOfAddresses === 0 ? 'No address found'
+      : `${numberOfAddresses}${numberOfAddresses === 1 ? ' address ' : ' addresses '}found`;
+  }
+
+  private setFormValue() {
```

Comment:
@johnbenjamin-hmcts - missing return type

Reply:
@Daniel-Lam - I have updated the code.